Add Microsoft DevAzure Server notes (#2851)
codenvy/che-docs: modules/end-user-guide/pages/using-a-git-provider-access-token.adoc

<1> Your {prod-short} user ID.
<2> The Git provider name: `github` or `gitlab` or `bitbucket-server` or `azure-devops`.
<3> The Git provider URL.

> <4> This line is only applicable to `azure-devops`: your Git provider user organization, or collection if DevAzure Server is used.

Reviewer: ```suggestion
<4> This line is only applicable to `azure-devops`: your Git provider user organization, or collection if Azure DevOps Server is used.
```

EOF
----
====

> . Azure DevOps Server. If you are using Azure DevOps Server, you must also modify the xref:mounting-git-configuration.adoc[workspace's

Reviewer: @vinokurig could you please make this block `IMPORTANT` similar to https://github.com/eclipse-che/che-docs/pull/2851/files#diff-ac7779248c50ba9a999fd70fb4eb27a90ee942e6709da76b96a988ff6b548ce1R39

Author: done

gitconfig] with the following section:
+

> ```gitconfig

Reviewer 2: is this a code block? just double-checking because it's not in asciidoc

Author: thanks, removed the leftover.

Reviewer 2: sorry, that's still not it. this is what asciidoc code blocks look like:

![image](https://github.com/user-attachments/assets/0d61f73a-2954-45e5-9884-e039b2c01ba5)

Author: I think the github branch was not updated, but it is fixed in the latest version.

----
====

[IMPORTANT]
====
If you are using Azure DevOps Server, you must also modify the xref:mounting-git-configuration.adoc[workspace's
gitconfig] with the following section:
```
[http]
    extraheader = "Authorization: Basic <base64-encoded(:personal-access-token)>"
```
To generate the key-value pair, use the following command:
```
echo -n "extraheader = \"Authorization: Basic "$(printf ":%s" <personal access token> | base64)\"
```
see link:https://learn.microsoft.com/en-us/azure/devops/repos/git/auth-overview?view=azure-devops&tabs=Linux#personal-access-tokens[the documentation page] for more information.

> This header is needed for remote git operations to Azure Devops Server, e.g. `git clone`. This authorization method has higher priority over the git credentials store, so git will fail on remote operations to other git providers. To be able to use other providers you need to remove the `extraheader` from the gitconfig.

Reviewer: ```suggestion
The `extraheader` configuration is needed for remote git operations to Azure Devops Server, e.g. `git clone`. This authorization method has a higher priority over the git credentials store, and as a result, the remote operations to other Git providers will fail.
```
